src/modules/keyboardmanager/KeyboardManagerEngineLibrary/KeyboardEventHandlers.cpp

```diff
@@ -32,6 +32,58 @@ namespace
     {
         return data->lParam->dwExtraInfo & CommonSharedConstants::KEYBOARDMANAGER_INJECTED_FLAG;
     }
+
+    void UpdateNumpadWithShift(LowlevelKeyboardEvent* data, State& state)
+    {
+        //Function for fixing numpad when used as shift https://github.com/microsoft/PowerToys/issues/22346
+        //VK_CLEAR is not encoded in IsNumpadOriginated
+        if (Helpers::IsNumpadOriginated(data->lParam->vkCode) || data->lParam->vkCode == VK_CLEAR)
+        {
+            // Decode it. If it is VK_CLEAR it will do nothing
+            DWORD decodedKey = Helpers::ClearKeyNumpadOrigin(data->lParam->vkCode);
+            //check if we already have a stored scanID
+            auto scanKey = MapVirtualKey(decodedKey, MAPVK_VK_TO_VSC);
+            auto it = state.scanMap.find(scanKey);
+            if (it != state.scanMap.end())
+            {
+                auto keyIt = state.GetSingleKeyRemap(it->second);
+                if (keyIt)
+                {
+                    //if key is stored as shift replace it with the numpad key
+                    auto keyValue = keyIt.value();
+                    if (keyValue->second.index() == 0)
+                    {
+                        auto key = std::get<DWORD>(keyValue->second);
+                        if (key == VK_LSHIFT || key == VK_RSHIFT)
```

Comment:
```suggestion
                        if (key == VK_LSHIFT || key == VK_RSHIFT || key == VK_SHIFT)
```
There's also a global shift VK which we need to catch. I've tried without this change and if we map "Numpad 1" to "Shift" instead of "Shift (Left)" or "Shift (Right)" the issue persists. This change might fix it.